lib/iris/tests/unit/fileformats/pp/test__data_bytes_to_shaped_array.py

```diff
@@ -17,10 +17,52 @@
 
 import numpy as np
 import numpy.ma as ma
+import pytest
 
 import iris.fileformats.pp as pp
 
 
+@pytest.mark.parametrize(
+    "data_shape,expected_shape,data_type",
+    [
+        ((2, 3), (2, 3), np.float32),
+        ((2, 3), (3, 2), np.int32),
+        ((2, 3), (1, 3), np.int16),
+        ((2, 3), (2, 2), np.int8),
+        ((2, 3), (3, 3), np.float32),
+        ((2, 3), (2, 4), np.float32),
+    ],
+)
+def test_data_padding__no_compression(data_shape, expected_shape, data_type):
+    data = np.empty(data_shape, dtype=data_type)
+
+    # create the field data buffer
+    buffer = io.BytesIO()
+    buffer.write(data)
+    buffer.seek(0)
+    data_bytes = buffer.read()
+
+    lbpack = pp.SplittableInt(0, dict(n1=0, n2=1))
+    boundary_packing = None
+    mdi = -1
+    args = (
+        data_bytes,
+        lbpack,
+        boundary_packing,
+        expected_shape,
+        data_type,
+        mdi,
+    )
+    data_length, expected_length = np.prod(data_shape), np.prod(expected_shape)
+
+    if expected_length <= data_length:
+        result = pp._data_bytes_to_shaped_array(*args)
+        assert result.shape == expected_shape
+    else:
+        with pytest.raises(ValueError):
```

Comment:
You know the specific `ValueError` you're expecting. Could you include a `match` argument to look for part of that specific message?

Reply:
Done :+1: